main.go

```diff
@@ -163,37 +164,48 @@ func StartPurgeLoop(secInterval int, chunkSize int64, expire time.Duration, omit
 	for range time.Tick(time.Duration(secInterval) * time.Second) {
 		job := instrument.NewJob("PumpRecordsPurge")
 
-		AnalyticsValues := AnalyticsStore.GetAndDeleteSet(storage.ANALYTICS_KEYNAME, chunkSize, expire)
-		if len(AnalyticsValues) > 0 {
-			startTime := time.Now()
+		iterationsPerLoop := 1
+		if SystemConfig.EnableMultipleAnalyticsKeys {
+			iterationsPerLoop = 10
+		}
+		for i := 0; i < iterationsPerLoop; i++ {
+			analyticsKeyName := storage.ANALYTICS_KEYNAME
+			if SystemConfig.EnableMultipleAnalyticsKeys {
+				analyticsKeyName += "_" + fmt.Sprint(i)
```

Comment:
I suggest to make it backward compatible, and read "old" key without prefix too. 
By doing so it will be safe upgrade pump first, and it will continue working with old gateways.

Reply:
@buger if `enable_multiple_analytics_keys` is false, it's backward compatible since it will only loop once https://github.com/TykTechnologies/tyk-pump/blob/8ded20fc66eafdef07f22089330fe091acdbe81b/main.go#L168.